From 8281f8f5255129e23f26356b6dc837dff1da0b84 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Vincent=20Harl=C3=A9?= <vincent.harle@softathome.com>
Date: Mon, 12 Dec 2022 17:34:15 +0100
Subject: [PATCH] [Debug] improve debug_bridge.sh to make it compatible with
 prplOs

---
 scripts/debug_bridge.sh | 30 ++++++++++++++++++++++--------
 1 file changed, 22 insertions(+), 8 deletions(-)

diff --git a/scripts/debug_bridge.sh b/scripts/debug_bridge.sh
index ac4ac9e..2165fff 100755
--- a/scripts/debug_bridge.sh
+++ b/scripts/debug_bridge.sh
@@ -9,20 +9,34 @@ fi
 
 . $LIBDEBUG
 
-BRIDGES="br-lan br-guest"
 
-dump_bridge()
+dump_bridge_full()
 {
+	#used for bridge-utils brctl
 	show_title "show bridge($1) information"
 	show_cmd brctl show $1
-	show_title "show bridge stp information"
 	show_cmd brctl showstp $1
-	show_title "show bridge macs information"
 	show_cmd brctl showmacs $1
 }
 
-for BRIDGE in $BRIDGES
-do
-	dump_bridge $BRIDGE
-done
+
+#detect if we have limited busybox brctl or bridge-utils brctl. crietria is showstp function
+if [ -z "$(brctl | grep showstp)" ] ; then
+	
+	#busybox brct is rather limited
+	show_title "show bridges information"
+	show_cmd brctl show
+else
+	#get bridges list from sysfs
+	BRIDGES=$(grep "DEVTYPE=bridge" /sys/class/net/*/uevent |sed -E  's|^/sys/class/net/(.*)/uevent.*$|\1|')
+
+	for BRIDGE in $BRIDGES
+	do
+		dump_bridge_full $BRIDGE
+	done
+
+fi 
+
+show_title "show arp information"
+show_cmd cat /proc/net/arp
 
-- 
GitLab
