From 90a857aab8bc5e42757e6543902c43b0eddb7154 Mon Sep 17 00:00:00 2001
From: Matthias FRANCK <matthias.franck_ext@softathome.com>
Date: Mon, 20 Nov 2023 12:06:44 +0100
Subject: [PATCH] Set CONFIG_SAH_AMX_TR181_XPON_USE_NETDEV_COUNTERS default to
 off

---
 odl/Makefile | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/odl/Makefile b/odl/Makefile
index f80a74b..0c8be66 100644
--- a/odl/Makefile
+++ b/odl/Makefile
@@ -1,24 +1,10 @@
-
 TARGETS=tr181-xpon.odl tr181-xpon_ANI.odl tr181-xpon_EthernetUNI.odl
 
 all: $(TARGETS)
 
-RUN_CONTAINER := $(or $(and $(wildcard /.dockerenv),1),0)
 M4_OPTS :=
 M4_DEPS :=
 
-ifeq ($(RUN_CONTAINER),1)
-include ../makefile.inc
-ifdef CONFIG_SAH_AMX_TR181_XPON_USE_NETDEV_COUNTERS
-M4_OPTS := -DCONFIG_SAH_AMX_TR181_XPON_USE_NETDEV_COUNTERS=y
-M4_DEPS := ../makefile.inc
-endif
-else
-include $(STAGINGDIR)/components.config
-M4_OPTS := -I$(STAGINGDIR)
-M4_DEPS := $(STAGINGDIR)/components.config
-endif
-
 %.odl: %.odl.m4 $(M4_DEPS)
 	m4 $(M4_OPTS) $< > $@
 
-- 
GitLab
