--- a/agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager.cpp
+++ b/agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager.cpp
@@ -1877,6 +1877,14 @@ void ApManager::handle_cmdu(ieee1905_1::
         if (perform_update && !bss_info_conf_list.empty()) {
             ap_wlan_hal->update_vap_credentials(bss_info_conf_list, backhaul_wps_ssid,
                                                 backhaul_wps_passphrase, bridge_name);
+#if defined(MORSE_MICRO)
+            //Update UCI wireless config
+            for (const auto &bss_it : bss_info_conf_list) {
+                if (!beerocks::bpl::bpl_cfg_set_wifi_credentials(m_iface, bss_it)) {
+                    LOG(ERROR) << "Could not set wireless credentials in UCI";
+                }
+            }
+#endif
 
             // hostapd is enabled and started radio beaconing after autoconfiguration
             // then radio tx power is updated, so to keep agent DB updated send CSA notification
--- a/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
+++ b/framework/platform/bpl/cfg/uci/bpl_cfg_wifi.cpp
@@ -56,6 +56,11 @@ static bool bpl_cfg_get_bss_configuratio
             return WSC::eWscAuth::WSC_AUTH_SAE;
 #endif
         }
+#if defined(MORSE_MICRO)
+        else if ("sae-mixed" == encryption) {
+            return WSC::eWscAuth(WSC::eWscAuth::WSC_AUTH_SAE | WSC::eWscAuth::WSC_AUTH_WPA2PSK);
+        }
+#endif
         return WSC::eWscAuth::WSC_AUTH_INVALID;
     };
     configuration.authentication_type = get_authentication_type(options["encryption"]);
@@ -69,6 +74,11 @@ static bool bpl_cfg_get_bss_configuratio
                    contains("+aes", encryption) || contains("+ccmp", encryption)) {
             return WSC::eWscEncr::WSC_ENCR_AES;
         }
+#if defined(MORSE_MICRO)
+        else if ("sae-mixed" == encryption) {
+            return WSC::eWscEncr::WSC_ENCR_AES;
+        }
+#endif
         return WSC::eWscEncr::WSC_ENCR_INVALID;
     };
     configuration.encryption_type = get_encryption_type(options["encryption"]);
@@ -406,6 +416,13 @@ bool bpl_cfg_set_wifi_credentials(const
         } else if (authentication_type == WSC::eWscAuth::WSC_AUTH_SAE) {
             encryption = "sae";
         }
+#if defined(MORSE_MICRO)
+        //Handle Mixed Mode
+        else if (authentication_type ==
+                   WSC::eWscAuth(WSC::eWscAuth::WSC_AUTH_WPA2PSK | WSC::eWscAuth::WSC_AUTH_SAE)) {
+            encryption = "sae-mixed";
+        }
+#endif
         return encryption;
     };
     options["encryption"] =
