commit 6ab5aee76613fa1a22cb6af26989500ed1dc7bf1
Author: Prabha Murugesan <prabha.murugesan@morsemicro.com>
Date:   Fri Jun 20 16:52:19 2025 +1000

    [APP-4739][PR #41] Reset controller discovery status

diff --git a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
index 426817326..b39952d57 100644
--- a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
+++ b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
@@ -251,6 +251,19 @@ void ApAutoConfigurationTask::handle_event(uint8_t event_enum_value, const void
 
             LOG(DEBUG) << "starting discovery sequence on radio_iface=" << radio->front.iface_name;
             FSM_MOVE_STATE(radio->front.iface_name, eState::CONTROLLER_DISCOVERY);
+
+            /*
+            * Reset discovery status to handle a rare race condition:
+            * - If START_AP_AUTOCONFIGURATION is received again during CONTROLLER_DISCOVERY
+            *   (e.g. due to backhaul reconnect), the FSM re-enters discovery with:
+            *     msg_sent = true, completed = false
+            * - This causes the fsm to get stuck and skip resending the auto config search.
+            */
+            for (auto &discovery_status : m_discovery_status) {
+                if (!discovery_status.second.completed)
+                    discovery_status.second.msg_sent = false;
+            }
+
             m_task_is_active = true;
         }
 
