Index: tr181-ethernet-v1.13.2/scripts/ethernet-manager.sh
===================================================================
--- tr181-ethernet-v1.13.2.orig/scripts/ethernet-manager.sh
+++ tr181-ethernet-v1.13.2/scripts/ethernet-manager.sh
@@ -1,4 +1,6 @@
 #!/bin/sh
+
+source /usr/lib/amx/scripts/amx_init_functions.sh
 [ -f /etc/environment ] && source /etc/environment
 ulimit -c ${ULIMIT_CONFIGURATION:-0}
 name="ethernet-manager"
@@ -9,12 +11,8 @@ case $1 in
         ${name} -D
         ;;
     stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill `cat /var/run/${name}.pid`
-        else
-            killall ${name}
-	fi
-        ;;
+        process_stop_shutdown ${name} 5
+	;;
     debuginfo)
         ubus-cli "Ethernet.?"
         ;;
