Index: tr181-time-v2.4.3/scripts/time-manager.sh
===================================================================
--- tr181-time-v2.4.3.orig/scripts/time-manager.sh
+++ tr181-time-v2.4.3/scripts/time-manager.sh
@@ -1,4 +1,6 @@
 #!/bin/sh
+
+source /usr/lib/amx/scripts/amx_init_functions.sh
 [ -f /etc/environment ] && source /etc/environment
 ulimit -c ${ULIMIT_CONFIGURATION:-0}
 name="time-manager"
@@ -8,12 +10,8 @@ case $1 in
         time-manager -D
         ;;
     stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill `cat /var/run/${name}.pid`
-        else
-            killall ${name}
-        fi
-        ;;
+        process_stop_shutdown ${name} 5
+	;;
     debuginfo)
         ubus-cli "Time.?"
         ;;
