From e37c6f9d708c454c802eede814cc51fb89c919e9 Mon Sep 17 00:00:00 2001
From: vdutta <vivek.dutta@iopsys.eu>
Date: Thu, 3 Feb 2022 17:14:51 +0530
Subject: [PATCH] Fix compilation issue with lws-4

---
 src/main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/main.c b/src/main.c
index 055e49663b12..4b75180b210f 100644
--- a/src/main.c
+++ b/src/main.c
@@ -557,7 +557,9 @@ ssl:
 	lws_info.user = &global;
 	lws_info.options = LWS_SERVER_OPTION_EXPLICIT_VHOSTS | (global_cfg.any_ssl ? LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT : 0);
 	lws_info.server_string = "owsd";
+#if LWS_LIBRARY_VERSION_NUMBER >= 2001000 && LWS_LIBRARY_VERSION_NUMBER < 4000000
 	lws_info.ws_ping_pong_interval = 300;
+#endif
 
 	lwsl_debug("Creating lwsl context\n");
 
