Index: timingila-v2.0.2/scripts/timingila-core.sh
===================================================================
--- timingila-v2.0.2.orig/scripts/timingila-core.sh
+++ timingila-v2.0.2/scripts/timingila-core.sh
@@ -1,27 +1,18 @@
 #!/bin/sh
+
+source /usr/lib/amx/scripts/amx_init_functions.sh
 [ -f /etc/environment ] && source /etc/environment
 ulimit -c ${ULIMIT_CONFIGURATION:-0}
 name="timingila"
 dm_name="Timingila"
 
-kill_pid()
-{
-    PIDFILE=${1}
-    if [ -f "${PIDFILE}" ]; then
-        kill `cat ${PIDFILE}`
-    else
-        echo "Can't find ${PIDFILE}. Killing all ${name} processes"
-        killall ${name}
-    fi
-}
-
 case $1 in
     start|boot)
         $name -D
         ;;
     stop|shutdown)
-        kill_pid /var/run/${name}.pid
-        ;;
+        process_stop_shutdown ${name} 5
+	;;
     debuginfo)
         ubus-cli "Timingila.?"
         ;;
