--- a/scripts/environment.sh
+++ b/scripts/environment.sh
@@ -17,7 +17,7 @@ prepare_environment() {
     MODELNUMBER=""
     DESCRIPTION=$(awk '$1 ~ /DISTRIB_DESCRIPTION/' /etc/openwrt_release | cut -d "'" -f2 | cut -d "'" -f1)
     HARDWAREVERSION=$(awk '$1 ~ /DISTRIB_TARGET/' /etc/openwrt_release | cut -d "'" -f2 | cut -d "'" -f1)
-    SOFTWAREVERSION=$(grep -oE 'Linux version [0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}' /proc/version | awk '{ print $NF }')
+    SOFTWAREVERSION=$(source /etc/os-release ; echo $VERSION_ID)
 
     if [ -f "/proc/environment/ethaddr" ]; then
         BASEMACADDRESS=$(cat /proc/environment/ethaddr)
