From 688c6f4448797b6d0b50f5a3544bb11c619db609 Mon Sep 17 00:00:00 2001
From: Matthias FRANCK <matthias.franck_ext@softathome.com>
Date: Tue, 1 Mar 2022 14:47:43 +0100
Subject: [PATCH] Add missing include dirs

---
 src/makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/makefile b/src/makefile
index 28bb43e..0a0d44c 100644
--- a/src/makefile
+++ b/src/makefile
@@ -15,8 +15,8 @@ TARGET_A = $(OBJDIR)/$(COMPONENT).a
 SRCDIR = .
 INCDIR_PUB = ../include
 INCDIR_PRIV = ../include_priv
-INCDIRS = $(INCDIR_PUB) $(INCDIR_PRIV) $(if $(STAGINGDIR), $(STAGINGDIR)/include)
-LIBDIR = $(if $(STAGINGDIR), -L$(STAGINGDIR)/lib -L$(STAGINGDIR)/usr/lib)
+INCDIRS = $(INCDIR_PUB) $(INCDIR_PRIV) $(if $(STAGINGDIR), $(STAGINGDIR)/include $(STAGING_DIR)/opt/prplos/usr/include)
+LIBDIR = $(if $(STAGINGDIR), -L$(STAGINGDIR)/lib -L$(STAGINGDIR)/usr/lib -L$(STAGING_DIR)/opt/prplos/usr/ -L$(STAGING_DIR)/opt/prplos/usr/lib/)
 
 # files
 HEADERS = $(wildcard $(INCDIR_PUB)/rlyeh/*.h)
-- 
GitLab

