From c02bc05fedd0e7765fd2c7289d6b68fd4950f7d5 Mon Sep 17 00:00:00 2001
From: Matthias FRANCK <matthias.franck_ext@softathome.com>
Date: Thu, 8 Sep 2022 11:10:32 +0200
Subject: [PATCH] Update ipv6 rules

---
 config/firewall_ipv6.defaults | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/config/firewall_ipv6.defaults b/config/firewall_ipv6.defaults
index efd0307..dd2b0c4 100644
--- a/config/firewall_ipv6.defaults
+++ b/config/firewall_ipv6.defaults
@@ -22,6 +22,19 @@
 -A FORWARD -j MINIUPNPD
 -A FORWARD -j FORWARD6_Firewall
 -A FORWARD6_First -i br-lan -o br-lan -j ACCEPT
+-I FORWARD6_First -m ipv6header --soft --header none -j ACCEPT
+-A FORWARD6_First -s fc00::/7 -j DROP
+-A FORWARD6_First -d fc00::/7 -j DROP
+-A FORWARD6_First -s fec0::/10 -j DROP
+-A FORWARD6_First -d fec0::/10 -j DROP
+-A FORWARD6_First -s ::ffff:0:0/96 -j DROP
+-A FORWARD6_First -d ::ffff:0:0/96 -j DROP
+-A FORWARD6_First -s ::0000:0:0/96 -j DROP
+-A FORWARD6_First -d ::0000:0:0/96 -j DROP
+-A FORWARD6_First -s 2001:db8::/32 -j DROP
+-A FORWARD6_First -d 2001:db8::/32 -j DROP
+-A FORWARD6_First -s 2001:10::/28 -j DROP
+-A FORWARD6_First -d 2001:10::/28 -j DROP
 -A INPUT6_First -i lo -j ACCEPT
 -A INPUT6_First -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
 -A INPUT6_First -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -j syn_flood
-- 
GitLab
