From 23d44fed79f9a2a5bd464612da1d5ab3896dad62 Mon Sep 17 00:00:00 2001
From: Alexander Van Parys <alexander.vanparys@softathome.com>
Date: Fri, 16 Dec 2022 11:03:09 +0100
Subject: [PATCH] Remove own implementation of when_true_status macro

---
 include_priv/firewall.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/include_priv/firewall.h b/include_priv/firewall.h
index f4fe9d6..090f17a 100644
--- a/include_priv/firewall.h
+++ b/include_priv/firewall.h
@@ -85,10 +85,6 @@ extern "C"
 #define OBJECT_NAME(OBJECT) amxd_object_get_name(OBJECT, AMXD_OBJECT_NAMED)
 #define STRING_EMPTY(TEXT) ((TEXT == NULL) || (*TEXT == 0))
 
-// macro from amxd
-#define when_true_status(x, l, c) \
-    if(x) { c; goto l; }
-
 #define INTERFACE_LEN 16
 #define IP_LEN 48
 
-- 
GitLab
