From 86b97dcef52b2ea151a62f8c3910ca755630ca59 Mon Sep 17 00:00:00 2001
From: sahbot <support.opensource@softathome.com>
Date: Tue, 14 Nov 2023 12:33:21 +0100
Subject: [PATCH] Issue: NET-5278 [AMX] Only detect sockets for loaded backends
 [fix]

GitOrigin-RevId: 4cbf45201a110d76a7e4cc44908b76e3fbd28ef2

diff --git a/src/amxrt_connections.c b/src/amxrt_connections.c
index f1fefc5c3fd3..0cb745912adf 100644
--- a/src/amxrt_connections.c
+++ b/src/amxrt_connections.c
@@ -184,25 +184,30 @@ static void amxrt_add_known_uris(amxc_var_t* uris) {
     struct stat sb;
 
     // always check if not already added
-    if(stat("/var/run/pcb_sys", &sb) == 0) {
-        if(!amxrt_list_contains(uris, "pcb:/var/run/pcb_sys")) {
-            amxc_var_add(cstring_t, uris, "pcb:/var/run/pcb_sys");
+    if(amxb_be_get_info("pcb") != NULL) {
+        if(stat("/var/run/pcb_sys", &sb) == 0) {
+            if(!amxrt_list_contains(uris, "pcb:/var/run/pcb_sys")) {
+                amxc_var_add(cstring_t, uris, "pcb:/var/run/pcb_sys");
+            }
         }
     }
-
-    if((stat("/var/run/ubus.sock", &sb) == 0) || (stat("/var/run/ubus/ubus.sock", &sb) == 0)) {
-        if(stat("/var/run/ubus.sock", &sb) == 0) {
-            if(!amxrt_list_contains(uris, "ubus:/var/run/ubus.sock")) {
-                amxc_var_add(cstring_t, uris, "ubus:/var/run/ubus.sock");
-            }
-        } else {
-            if(!amxrt_list_contains(uris, "ubus:/var/run/ubus/ubus.sock")) {
-                amxc_var_add(cstring_t, uris, "ubus:/var/run/ubus/ubus.sock");
+    if(amxb_be_get_info("ubus") != NULL) {
+        if((stat("/var/run/ubus.sock", &sb) == 0) || (stat("/var/run/ubus/ubus.sock", &sb) == 0)) {
+            if(stat("/var/run/ubus.sock", &sb) == 0) {
+                if(!amxrt_list_contains(uris, "ubus:/var/run/ubus.sock")) {
+                    amxc_var_add(cstring_t, uris, "ubus:/var/run/ubus.sock");
+                }
+            } else {
+                if(!amxrt_list_contains(uris, "ubus:/var/run/ubus/ubus.sock")) {
+                    amxc_var_add(cstring_t, uris, "ubus:/var/run/ubus/ubus.sock");
+                }
             }
         }
     }
 
-    amxp_dir_scan("/var/run/usp", "d_type == DT_SOCK", false, amxrt_connection_add_usp_socket, uris);
+    if(amxb_be_get_info("usp") != NULL) {
+        amxp_dir_scan("/var/run/usp", "d_type == DT_SOCK", false, amxrt_connection_add_usp_socket, uris);
+    }
 }
 
 static void amxrt_set_backend_setting(amxc_var_t* config,
